cmd/dfget/app/root.go

```diff
@@ -0,0 +1,217 @@
+package app
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"strconv"
+	"strings"
+	"time"
+
+	cfg "github.com/dragonflyoss/Dragonfly/dfget/config"
+	"github.com/dragonflyoss/Dragonfly/dfget/core"
+	"github.com/dragonflyoss/Dragonfly/dfget/errors"
+	"github.com/dragonflyoss/Dragonfly/dfget/util"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+var (
+	localLimit string
+	totalLimit string
+	filter     string
+)
+
+var rootCmd = &cobra.Command{
+	Use:   "dfget",
+	Short: "The dfget is the client of Dragonfly.",
+	Long:  "The dfget is the client of Dragonfly, a non-interactive P2P downloader.",
+	Run: func(cmd *cobra.Command, args []string) {
+		checkParameters()
+		initLog()
+		cfg.AssertContext(cfg.Ctx)
+		cfg.Ctx.ClientLogger.Infof("cmd params:%q", args)
+		initProperties()
+		cfg.Ctx.ClientLogger.Infof("context:%s", cfg.Ctx)
+		err := core.Start(cfg.Ctx)
+		if err != nil {
+			util.Printer.Println(resultMsg(cfg.Ctx, time.Now(), err))
+			os.Exit(err.Code)
+		}
+		util.Printer.Println(resultMsg(cfg.Ctx, time.Now(), err))
+		os.Exit(0)
```

Comment:
Extract the above code as a separate function. It's just my personal advice. 😄

Reply:
Your mean: separate `util.Printer.Println(resultMsg(cfg.Ctx, time.Now(), err))` to 
```go
resultMsg := resultMsg(cfg.Ctx, time.Now(), err)
util.Printer.Println(resultMsg)
```